src/storage.js

```diff
@@ -0,0 +1,430 @@
+  var action_map = getBadgerStorageObject('action_map');
+  var relevantDomains = [];
+  for(var domain in action_map.getItemClones()){
+    if(selector == getActionForFqdn(domain)){
+      relevantDomains.push(domain); 
+    }
+  }
+  return relevantDomains;
+};
+
+/**
+ * Get the number of domains that the given FQDN has been seen tracking on
+ * @param fqdn domain to check status of
+ * @return int the number of domains fqdn has been tracking on 
+ */
+var getTrackingCount = function(fqdn){
+  var snitch_map = getBadgerStorageObject('snitch_map');
+  if(snitch_map.hasItem(fqdn)){
+    return snitch_map.getItem(fqdn).length;
+  } else {
+    return 0;
+  }
+};
+
+/**
+ * A factory for getting BadgerStorage objects, this will either get a badger 
+ * storage object from the cache or return a new BadgerStorage object. 
+ * @param {String} key the name of the stored object
+ * @return {BadgerStorage} A badgerStorage object 
+ **/
+ 
+var getBadgerStorageObject = function(key) {
+
+  if(badgerPen.hasOwnProperty(key)){
+    return badgerPen[key];
+  }
+  console.error('cant initialize cache from getBadgerStorageObject. You are using this API improperly');
+};
+
+var _initializeCache = function(keys, cb) {
+
+  // now check localStorage
+  chrome.storage.local.get(keys, function(store){
+    _.each(keys, function(key){
+      if(store.hasOwnProperty(key)){
+        badgerPen[key] = new BadgerStorage(key, store[key]);
+      } else {
+        var storage_obj = new BadgerStorage(key, {});
+        badgerPen[key] = storage_obj;
+        _syncStorage(storage_obj);
+      }
+    });
+    if(_.isFunction(cb)){
+      cb();
+    }
+  });
+};
+
+/**
+ * Privacy Badger Storage Object. Has methods for getting, setting and deleting
+ * should be used for all storage needs, transparently handles data presistence
+ * syncing and private browsing.
+ * Usage:
+ * example_map = getBadgerStorageObject('example_map');
+ * # instance of BadgerStorage
+ * example_map.setItem('foo', 'bar') 
+ * # null
+ * example_map
+ * # { foo: "bar" }
+ * example_map.hasItem('foo')
+ * # true
+ * example_map.getItem('foo');
+ * # 'bar'
+ * example_map.getItem('not_real');
+ * # undefined
+ * example_map.deleteItem('foo');
+ * # null
+ * example_map.hasItem('foo');
+ * # false
+ * 
+ **/
+
+/**
+ * BadgerStorage constructor
+ * *DO NOT USE DIRECTLY* Instead call `getBadgerStorageObject(name)`
+ * @param {String} name the name of the storage object
+ * @param {Object} seed the base object which we are instantiating from
+ * @return {BadgerStorage} an existing BadgerStorage object or an empty new object
+ **/
+var BadgerStorage = function(name, seed){
+  this.name = name;
+  this._store = seed;
+};
+
+BadgerStorage.prototype = {
+
+  /**
+   * check if this storage object has an item 
+   * @param {String} key the key for the item
+   * @return boolean 
+   **/
+  hasItem: function(key){
+    var self = this;
+    return !!self._store.hasOwnProperty(key);
```

Comment:
I'm a bit of a JS newb, is there a need to coerce the result of *hasOwnProperty* to a boolean? Is there a situation where the result would be something other than a boolean?